src/prefect/exceptions.py

```diff
@@ -349,3 +349,7 @@ class NotPausedError(PrefectException):
 
 class FlowPauseTimeout(PrefectException):
     """Raised when a flow pause times out"""
+
+
+class FlowPauseExit(PrefectException):
```

Comment:
Why did you decide to use exception raising vs. `return` statements here?

Reply:
The engine needs a signal to gracefully stop execution immediately, so I decided to raise an exception that's caught instead of waiting until the next task. The exception makes the downstream handling pretty convenient, e.g. knowing it's okay to leave the flow without a terminal state etc

Reply:
Without an exception, the user would need to check the return value of the pause function and exit themselves to prevent additional code from being executed.